src/core/ext/xds/xds_client.cc

```diff
@@ -561,17 +556,17 @@ XdsClient::ChannelState::RetryableCall<T>::RetryableCall(
                    .set_jitter(GRPC_XDS_RECONNECT_JITTER)
                    .set_max_backoff(Duration::Seconds(
                        GRPC_XDS_RECONNECT_MAX_BACKOFF_SECONDS))) {
-  // Closure Initialization
-  GRPC_CLOSURE_INIT(&on_retry_timer_, OnRetryTimer, this,
-                    grpc_schedule_on_exec_ctx);
   StartNewCallLocked();
 }
 
 template <typename T>
 void XdsClient::ChannelState::RetryableCall<T>::Orphan() {
   shutting_down_ = true;
   calld_.reset();
-  if (retry_timer_callback_pending_) grpc_timer_cancel(&retry_timer_);
+  if (timer_handle_.has_value()) {
+    GetDefaultEventEngine()->Cancel(*timer_handle_);
```

Comment:
This was intentional.  As you say, in this case, it doesn't matter whether cancellation succeeds, because the callback is a no-op if we cancel it, so I didn't bother checking the result.

Given that there are cases like these where it doesn't matter if cancellation succeeds, I'm not sure that `GRPC_MUST_USE_RESULT` would make sense here.  But on the other hand, both of these cases *could* check the result without too much trouble if we did use `GRPC_MUST_USE_RESULT`.  I think let's wait and see if there are other callers where this kind of thing actually becomes a bug.